enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,256 @@
+    following characteristics:
+    - `microshift-lvms-snapshotting` will depend on `microshift-lvms`.
+    - `microshift-lvms` will depend on `lvm2` to ensure host compatibility
+    - `microshift-lvms` will encapsulate LVMS and CSI driver container images and manifests. A post-install script 
+      will attempt a best-guess at a default volume-group by reimplementing the go log in microshift as bash or 
+      python. Not being able to identify a suitable volume-group is not a fatal error, and a warning will be logged 
+      during install. The user will be responsible for properly configuring the lvmd file.
+    - `microshift-lvms-snapshotting` is effectively a wrapper around CSI snapshotting images and manifests. It does 
+      not encapsulate LVMS images or configurations.
+  - It is critical that user-defined LVMD configurations be preserved during upgrades, downgrades, and reinstalls. RPM 
+    post-install logic will be not stomp on existing lvmd configurations. Overwriting the lvmd config poses a 
+    significant threat of orphaning LVM volumes if MicroShift were to change the volume-groups used by lvmd.
+
+### Risks and Mitigations
+
+- N/A
+
+### Drawbacks
+
+- N/A
+
+## Test Plan
+
+There are already tests for RPM and Ostree install processes. This proposal only slightly alters the installation 
+process and thus will require only minimal changes to existing tests.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- End-to-end tests
+- Sufficient time for feedback
+- Available by default
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Upgrade / Downgrade Strategy
+
+LVMS and CSI operate on existing Kubernetes APIs to track the state of cluster storage. It is safe to upgrade or 
+downgrade the runtime components without losing accountability or otherwise endangering provisioned storage.
+
+## Version Skew Strategy
+
+Risk of version skew is mitigated because the LVMS and CSI components are built from the same release image as the 
+rest of MicroShift's components. It is safe to upgrade and downgrade LVMS and/or CSI within the same major version 
+of MicroShift, just as it is on OCP.
+
+## Operational Aspects of API Extensions
+
+- N/A
+
+## Support Procedures
+
+- N/A
+
+## Alternatives
+
+Use the MicroShift or LVMD config to determine deployment of LVMS and/or snapshotting. It does not satisfy the goal of
+optionalizing the installation of the components. Instead, this solution would simply enable/disable the components.
```

Comment:
Could you quickly explain why a MicroShift config flag would not optionalize the installation of the components (maybe Im just missing something here)

Reply:
Using a flag to optionalize storage has some drawbacks (off the top of my head):
- MicroShift carries some tech-debt for rebasing, deploying, and configuring LVMS.
- treating lvms as a first class citizen in microshift isn't necessary. it's just a cluster workload like any other CSI driver.
- should the default be 'on' or 'off'? what should happen if the toggle is switched from 'on' to 'off'? How ever these are answered means more responsibility for microshift, not less.
- disk consumption is another consideration. for offline deployments, should microshift still include the lvms images just in case users switch it on after starting microshift?

I think the most compelling point is that by decoupling LVMS from MicroShift, removing it's first-class status, we're not just giving users ability to make the choice during ostree build time, we're also laying out groundwork for how users could install other CSI drivers, should the time come to support other CSI drivers.